ladybug/datacollection.py

```diff
@@ -1,447 +1,438 @@
+    def filter_by_hoys(self, hoys):
+        """Filter the Data Collection based on an analysis period.
 
         Args:
-            data: A list of DataPoint to be processed
-            month_range: A list of numbers for months. Default is 1-12
-        """
-        hourly_data_by_month = OrderedDict()
-        for m in month_range:
-            hourly_data_by_month[m] = []
-
-        for d in data:
-            try:
-                hourly_data_by_month[d.datetime.month].append(d)
-            except KeyError:
-                # month is not there
-                pass
-
-        return hourly_data_by_month
+           hoys: A List of hours of the year 0..8759
 
-    def group_by_month(self, month_range=xrange(1, 13)):
+        Return:
+            A new Data Collection with filtered data
         """
-        Return a dictionary of data points that are grouped for each month.
+        _moys = tuple(int(hour * 60) for hour in hoys)
+        return self.filter_by_moys(_moys)
 
-        Key values are between 1-12
+    def filter_by_moys(self, moys):
+        """Filter the Data Collection based on a list of minutes of the year.
 
         Args:
-           month_range: A list of numbers for months. Default is 1-12
-
-        Usage:
+           moys: A List of minutes of the year [0..8759 * 60]
 
-           epwfile = EPW("epw file address")
-           monthly_values = epwfile.dry_bulb_temperature.group_by_month()
-           print(monthly_values[2]) # returns values for the month of March
+        Return:
+            A new Data Collection with filtered data
         """
-        return self.group_data_by_month(self.data, month_range)
+        _filt_values, _filt_datetimes = self._filter_by_moys_slow(moys)
+        return HourlyDiscontinuousCollection(
+            self.header.duplicate(), _filt_values, _filt_datetimes)
 
-    @staticmethod
-    def group_data_by_day(data, day_range=xrange(1, 366)):
-        """
-        Return a dictionary of data points that are grouped by each day of year.
+    def group_by_day(self):
+        """Return a dictionary of this collection's values grouped by each day of year.
 
-        Key values are between 1-365
+        Key values are between 1-365.
+        """
+        data_by_day = OrderedDict()
+        for d in xrange(1, 366):
+            data_by_day[d] = []
+        for v, dt in zip(self._values, self.datetimes):
+            data_by_day[dt.doy].append(v)
+        return data_by_day
+
+    def average_daily(self):
+        """Return a daily collection of values averaged for each day."""
+        data_dict = self.group_by_day()
+        avg_data, d_times = [], []
+        for i in self.header.analysis_period.doys_int:
+            vals = data_dict[i]
+            if vals != []:
+                avg_data.append(sum(vals)/len(vals))
+                d_times.append(i)
+        new_header = self.header.duplicate()
+        new_header.metadata['statistical operation'] = 'Average'
```

Comment:
```suggestion
        new_header.metadata['statistical_operation'] = 'Average'
```

Reply:
You are right that the underscore should be used but, after thinking about this more, I think I can just simplify it to "operation" instead of "statistical_operation" since I can't think of any other type of metadata that would need to use this name.  This will generally make things cleaner, including any displays of the metadata on charts.